apps/condo/domains/ticket/tasks/closeCompletedTickets.js

```diff
@@ -1,46 +1,103 @@
         status: { id: STATUS_IDS.COMPLETED },
         statusUpdatedAt_lte: weekAgo,
         deletedAt: null,
-    })
-
-    for (const ticket of ticketsToChange) {
-        try {
-            await Ticket.update(adminContext, ticket.id, {
-                status: { connect: { id: STATUS_IDS.CLOSED } },
-                dv: 1,
-                sender: { fingerprint: 'auto-close', dv: 1 },
-            })
-        } catch (error) {
-            taskLogger.error({
-                msg: 'Failed to close Ticket',
-                data: { id: ticket.id },
-                error,
-            })
+    }
+
+    const countTicketToChange = await Ticket.count(context, ticketWhere)
+    const countChangedTicketByOrganization = {}
+    let changedTicketCounter = 0
+    let skippedTicket = 0
+
+    while (countTicketToChange > changedTicketCounter) {
+        const ticketsToChange = await Ticket.getAll(context, ticketWhere, {
+            first: CHUNK_SIZE,
+            // NOTE The list needs to be sorted in the same order all the time.
+            // This is necessary so that we can skip the required number of elements without allowing repetitions
+            sortBy: ['createdAt_ASC', 'createdBy_ASC'],
+            // NOTE It is necessary to skip tickets that could not be updated
+            // or were skipped in order to avoid looping on the same elements
+            skip: skippedTicket,
+        })
+
+        if (isEmpty(ticketsToChange)) break
+
+        changedTicketCounter += ticketsToChange.length
+
+        for (const ticket of ticketsToChange) {
+            const organizationId = get(ticket, 'organization.id', null)
+            if (countChangedTicketByOrganization[organizationId] && countChangedTicketByOrganization[organizationId] >= limitByOrganization) {
+                skippedTicket += 1
+                continue
+            }
+
+            try {
+                const updatedData = {
+                    dv: 1,
+                    sender: { fingerprint: 'auto-close', dv: 1 },
+                    status: { connect: { id: STATUS_IDS.CLOSED } },
+                }
+
+                await Ticket.update(context, ticket.id, updatedData)
+
+                if (organizationId in countChangedTicketByOrganization) {
+                    countChangedTicketByOrganization[organizationId] += 1
+                } else {
+                    countChangedTicketByOrganization[organizationId] = 1
+                }
+            } catch (error) {
+                skippedTicket += 1
+                taskLogger.error({
+                    msg: 'Failed to close Ticket',
+                    data: { id: ticket.id },
+                    error,
+                })
+            }
         }
     }
 }
 
 module.exports = {
-    closeCompletedTickets: createCronTask('closeCompletedTickets', '0 1 * * *', closeCompletedTickets),
+    closeCompletedTicketsCron: createCronTask('closeCompletedTickets', '0 1 * * *', closeCompletedTickets),
```

Comment:
You can call this task not only from corn. You don't need to rename it

Reply:
but we call `createCronTask`...

We do separate `closeCompletedTickets ` function import to test the it.